src/world/action/use_spinning_flower.c

```diff
@@ -1,5 +1,280 @@
+            } else {
+                D_802B6ED0 = sp20 & 0x3FF;
+            }
+        } else {
+            D_802B6ED0 = -1;
+        }
+        if (!(playerStatus->animFlags & (PLAYER_ANIM_FLAG_HOLDING_ITEM | PLAYER_ANIM_FLAG_2))) {
+            suggest_player_anim_clearUnkFlag(0x1002B);
+        }
+    }
+    switch(playerStatus->fallState) {
+        case 1:
+            gOverrideFlags |= 0x40;
+            if (++D_802B6EE4 >= 20.0f) {
+                D_802B6EE4 = 20.0f;
+            }
+            playerStatus->spriteFacingAngle = clamp_angle(playerStatus->spriteFacingAngle + D_802B6EE4);
+            if (playerStatus->framesOnGround < 10) {
+                playerStatus->framesOnGround++;
+                D_802B6EF4++;
+            }
+            D_802B6EE8 += 8.0f;
+            playerStatus->position.y = D_802B6EF4 + sin_rad(clamp_angle(D_802B6EE8) * 6.28318f / 360.0f) * 4.0f;
+            if (D_802B6ED0 >= 0) {
+                entityByIndex = get_entity_by_index(D_802B6ED0);
+                tempDistance = dist2D(entityByIndex->position.x, entityByIndex->position.z, playerStatus->position.x, playerStatus->position.z);
+                D_802B6EEC = atan2(entityByIndex->position.x, entityByIndex->position.z, playerStatus->position.x, playerStatus->position.z);
+                if (tempDistance > 4.0f) {
+                    tempDistance--;
+                }
+                sin_cos_rad(D_802B6EEC * 6.28318f / 360.0f, &sp10, &sp14);
+                playerStatus->position.x = entityByIndex->position.x + (sp10 * tempDistance);
+                playerStatus->position.z = entityByIndex->position.z - (sp14 * tempDistance);
+                sin_cos_rad((D_802B6EEC - 91.0f) * 6.28318f / 360.0f, &sp10, &sp14);
+                D_802B6ED4 = sp10 * D_802B6EDC;
+                D_802B6ED8 = -sp14 * D_802B6EDC;
+            }
+            game_input_to_move_vector(&sp18, &sp1C);
+            if (sp1C != 0.0f) {
+                sin_cos_rad((sp18 * 6.28318f) / 360.0f, &sp10, &sp14);
+                sp1C *= 0.03125f;
+                if (sp1C < 0.1) {
+                    sp1C = 0.1f;
+                }
+                playerStatus->position.x += sp10 * sp1C;
+                playerStatus->position.z -= sp14 * sp1C;
+            }
+            gCameras->targetPos.x = playerStatus->position.x;
+            gCameras->targetPos.y = playerStatus->position.y;
+            gCameras->targetPos.z = playerStatus->position.z;
+            sp20 = func_802B6000_E29470();
+            if (sp20 < 0 || !(sp20 & 0x4000)) {
+                playerStatus->framesOnGround = 20;
+                D_802B6EE8 = 0.0f;
+                D_802B6EF4 = playerStatus->position.y;
+                playerStatus->fallState++;
+                D_802B6EF0 = 1.6f;
+                playerStatus->flags |= 0x800000;
+            }
+            if (gGameStatusPtr->pressedButtons & 0x00002000 && !(playerStatus->animFlags & (PLAYER_ANIM_FLAG_HOLDING_ITEM | PLAYER_ANIM_FLAG_2))) {
```

Comment:
There's a button enum in enums.h I think - you may wanna use the actual button value here